src/common/visualization/visualization.py

```diff
@@ -0,0 +1,81 @@
+# SPDX-FileCopyrightText: 2023 spdx contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+from typing import Dict, List
+
+try:
+    from networkx import DiGraph
+except ImportError:
+    DiGraph = None
+from spdx.document_utils import get_contained_spdx_element_ids, get_element_from_spdx_id
+from spdx.model.document import Document
+from spdx.model.relationship import Relationship
+
+
+def visualize_document(document: Document, visualize: str) -> None:
+    try:
+        from networkx.drawing import nx_agraph
+    except ImportError:
+        raise ImportError
+
+    graph = generate_graph_from_spdx(document)
+    _color_nodes(graph)
+    attributes_graph = nx_agraph.to_agraph(graph)  # convert to a graphviz graph
+    attributes_graph.draw(visualize, prog="dot")
+
+
+def generate_graph_from_spdx(document: Document) -> DiGraph:
+    try:
+        from networkx import DiGraph
+    except ImportError:
+        raise ImportError
+    graph = DiGraph()
+    graph.add_node(document.creation_info.spdx_id, element=document.creation_info)
+
+    contained_elements: List[str] = get_contained_spdx_element_ids(document)
+    contained_element_nodes = [
+        (spdx_id, {"element": get_element_from_spdx_id(document, spdx_id)}) for spdx_id in contained_elements
```

Comment:
Given that you quite often call `get_element_from_spdx_id` in the code below, I think it would be beneficial for performance if we introduced another `document_utils` method which returns a dictionary that contains all SPDX Ids mapped to their corresponding element. That is, it returns something like the payload in SPDX3.